### PR TITLE
Feature webform styling

### LINKED ISSUE #3486: Webform fields don't respect field width settings
# Story

As SiteNow webform creator, I would like my form fields to respect the width I set, so that we don't have very large fields when the input only accepts a small number of characters.

# Additional Info
An example of this https://admissions.prod.drupal.uiowa.edu/biomedical-sciences-supplemental-application. The ACT and SAT fields have a size of 15 defined but are showing as 100% width.

# Proposed solution

### LINKED ISSUE #4954: As a site owner, I can create webforms that have better typography hierarchy and spacing
# Story

As a site owner, I can create webforms that have better font sizes and spacing so that my forms are more aesthetically pleasing.  

# Proposed solution

- Create [spacing and typography updates](https://tippiestudents.prod.drupal.uiowa.edu/undergraduates/get-involved/tippie-top-21/submit-tippie-top-21-nomination) suggested in https://user-images.githubusercontent.com/1036433/157935745-73d09adc-9544-4a8b-a9c0-c26c7f51e200.png.  
  - Work with https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/scss/components/form/forms.scss and https://github.com/uiowa/uiowa/blob/main/docroot/themes/custom/uids_base/scss/webforms/webforms.scss
    - Move items with `webform` class from `forms.scss` into `webform.scss`. 
      - Override core form styles from `forms.scss` with `webform` prefix in `webform.scss`.  There are a lot of other pages that rely on `forms.scss` so it is important that we create new styles under the `webform` class. 
-  Allow users to set widths on their form fields: https://github.com/uiowa/uiowa/issues/3486#issuecomment-1090720811 
- Regarding https://github.com/uiowa/uiowa/issues/4954#issuecomment-1087995362, we want to keep this functionality and possibly design this to make less of a visual impact?

<details>
**Before:**

Forms Mockup Orig

**After:**

Forms Mockup-Updates
</details>

### LINKED ISSUE #5002: Re-scope submit button CSS selector to avoid namespace collisions for webform fields
# Problem
Targeting the id of `form_submit` in our `forms.scss` can create issues if a webform field key contains the word "submit"

# Steps to reproduce
Create a text field on a webform with the name "Submitter name"

# Proposed solution
Update the scope of the CSS selector in `forms.scss` so that it will only apply to `type="submit"`

### PR DESCRIPTION
**Pull request by bewoldt**
Resolves #4954 
Resolves https://github.com/uiowa/uiowa/issues/5002
Resolves https://github.com/uiowa/uiowa/issues/3486

# How to test

**For tippie form example**
1. Pull down tippie.students
`blt ds --site=students.tippie.uiowa.edu`
2. visit the page: https://tippiestudents.uiowa.ddev.site/undergraduates/get-involved/tippie-top-21/submit-tippie-top-21-nomination
3. Check to see if it matches the notes referenced in #4954.

**To test all available form fields**
1. Pull down sandbox prod
`blt ds --site=sandbox.uiowa.edu`
2. Test the page https://sandbox.uiowa.ddev.site/form

**Test bug fixes for https://github.com/uiowa/uiowa/issues/5002 and https://github.com/uiowa/uiowa/issues/3486**
- Add a text field and for the Title use "Submitter's Name" or anything with the word submit.
- Set a fixed width for a text field and verify that the field width is not being set to 100% if there is a value besides 60 in the field.

### CODE REVIEW

**Comment by joewhitsitt** [comment]
Is this ready for review @bewoldt?

Update from standup: It is.

**Comment by GaryRidgway** [comment]
I will pull down and test this

**Review by GaryRidgway — APPROVED** [review]
Everything looked great except for these icons on colored backgrounds:
Screen Shot 2022-04-13 at 2 42 56 PM

**Review by GaryRidgway — APPROVED** [review]
Actually, I think it wouldn't be too hard to change those to have a transparent background instead of a white one, can we make that change first and then I'll come back and approve again?

**Review by GaryRidgway — CHANGES_REQUESTED** [review]
I meant to request changes for the above approval.

**Comment by bewoldt** [comment]
I added a fix for the glyph horizontal rule.

**Comment by bspeare** [comment]
I'm also testing a few things out here on a few sites.

File: docroot/themes/custom/uids_base/scss/webforms/webforms.scss [hunk 1]
```diff
@@ -1,12 +1,226 @@
...
+  // Flexbox
+  // --------------------------------
+  .webform-flexbox {
+    & .webform-flex--container {
+      & .form-actions [type="submit"] {
+        width: 100%; // Set button full-width.
+        height: 57px;
+      }
+    }
+    & .form-item {
```
Comment (bspeare) [inline docroot/themes/custom/uids_base/scss/webforms/webforms.scss]: @bewoldt can you take a look at these flexbox layouts on https://sandbox.uiowa.ddev.site/webform-layouts and try removing the `.form-item` lines and look at the spacing issue?

**Comment by bspeare** [comment]
Should we try to fix this issue in this?  https://github.com/uiowa/uiowa/issues/3486#issuecomment-1090720811.

**Comment by bspeare** [comment]
The style updates look great!  Maybe add `.5rem` margin under these `<legend>` items to get a little more space?

Screen Shot 2022-04-13 at 3 31 03 PM

**Comment by bspeare** [comment]
Also, @bewoldt do you remember how to recreate this issue that you were having with the field turning gold?  https://github.com/uiowa/uiowa/issues/5002

**Comment by bewoldt** [comment]
> 

Add a text field and for the Title use "Submitter's Name" or anything with the word submit.

**Comment by bspeare** [comment]
I updated the selector that was making the field gold to resolve https://github.com/uiowa/uiowa/issues/5002.

**Review by GaryRidgway — APPROVED** [review]

**Comment by GaryRidgway** [comment]
Just re-reviewed, looks good to me!

**Pull request merged**